### DISCUSSION TITLE
Setup/ demo code for st7701s with esp32s3

### DISCUSSION THREAD

**Discussion by CosmicProgrammer840** [moononournation/Arduino_GFX#164] [Q&A] [answered]
any way you could help with setting up code for this display using a esp32s3
https://www.buydisplay.com/480x480-round-ips-tft-lcd-display-2-1-inch-capactive-touch-circle-screen
cant seem to get the display class working in arduino isp

**Comment by CosmicProgrammer840** [comment]
still no change on the display. Every uplod of hello world does nothing. What do I need to change in the src files to get this to work? I used provided esp32rgbpanel bus class and arduino st7701 rgb panel display class. attached is the start up code the vendor sent to me. I appened my display class .cpp file.

I am using esp32-s3-devkitM-1

[2.1 480x480 st7701s ic.txt](https://github.com/moononournation/Arduino_GFX/files/8921308/2.1.480x480.st7701s.ic.txt)

**Reply by moononournation** [reply]
What is the reason you develop your own class?

**Reply by CosmicProgrammer840** [reply]
i just implimented the manufactures start up code, still cant seem to get it working though.

**Reply by moononournation** [reply]
I don't get it, the original code can init ST7701. If you cannot turn on your display, you should check the connection first.

**Reply by CosmicProgrammer840** [reply]
hmm ill put if back to original and check connections i was under the impression that i needed the manf start up code because this is a st7701s IC not a st7701

**Reply by CosmicProgrammer840** [reply]
ok so I verified all my wiring and used a fresh download of the library. still nothing on my hello world sketch. can you confirm esp32s3m-1 and st7701s will work
[hello world sketch.txt](https://github.com/moononournation/Arduino_GFX/files/8941989/hello.world.sketch.txt)

**Reply by moononournation** [reply]
can you take a photo of your connection?

**Reply by CosmicProgrammer840** [reply]
this is how I have it wired, hard to get photos that easily show but I have triple checked connections including continuity.


[PINOUT - Sheet1.pdf](https://github.com/moononournation/Arduino_GFX/files/8950078/PINOUT.-.Sheet1.pdf)

**Reply by moononournation** [reply]
can you follow the default pin connection first?

**Reply by CosmicProgrammer840** [reply]
I have it pinned corisponding to the bus class listed for esp32s3 and st7701. I made that pdf to keep track of all my connections but used this to make it.
  39 /* CS */, 48 /* SCK */, 47 /* SDA */,
  18 /* DE */, 17 /* VSYNC */, 16 /* HSYNC */, 21 /* PCLK */,
  4 /* R0 */, 3 /* R1 */, 2 /* R2 */, 1 /* R3 */, 0 /* R4 */,
  10 /* G0 */, 9 /* G1 */, 8 /* G2 */, 7 /* G3 */, 6 /* G4 */, 5 /* G5 */,
  15 /* B0 */, 14 /* B1 */, 13 /* B2 */, 12 /* B3 */, 11 /* B4 */
); 
thank you for all your help

**Reply by CosmicProgrammer840** [reply]
along with pinning r5 with r4 and b5 with b4 to make it rgb 565

**Comment by CosmicProgrammer840** [comment]
here is some photos of my wiring. please tell me what you think.
thank you!
![20220621_165627](https://user-images.githubusercontent.com/96497670/174896562-e02f10a6-5949-4f36-9915-e0a7b0bf4dae.jpg)
![20220621_165703](https://user-images.githubusercontent.com/96497670/174896565-4487b366-bc6b-4a71-be53-2c1a16e33faa.jpg)
![20220621_165722](https://user-images.githubusercontent.com/96497670/174896571-dc5f2511-758f-41d3-8393-b0f4badf5d34.jpg)
![20220621_165618](https://user-images.githubusercontent.com/96497670/174896576-d03eb39d-0e23-4755-b3db-a10c5512514d.jpg)
![20220621_165558](https://user-images.githubusercontent.com/96497670/174896580-933255b4-1530-4f6e-b23c-08143f38e47a.jpg)
s

**Comment by CosmicProgrammer840** [comment]
![ESP32-S3-DevKitM-1_L](https://user-images.githubusercontent.com/96497670/174898073-f9d61039-6f3d-433a-9016-158a64abd48b.png)

**Comment by CosmicProgrammer840** [comment]
found out the issue still working on solving it, heres the debug info
defult pinout and default hello world sketch

[debug.txt](https://github.com/moononournation/Arduino_GFX/files/8959217/debug.txt)

**Reply by moononournation** [reply]
enable psram

**Reply by CosmicProgrammer840** [reply]
these are my settings any suggestions?

upload mode: uart0/hardwarecdc
cpu freq : 80mhz
flash mode : qio 80mhz
flash size : 8MB (64mb)
partition scheme : 8M (3MB APP/ 1.5MB FAT)
PSRAM : QSPI PSRAM
arduino runs on : core 1
Events run on : core 1

[debug.txt](https://github.com/moononournation/Arduino_GFX/files/8959490/debug.txt)

**Comment by CosmicProgrammer840** [comment]
hello world sketch still only gives me garbled image. fades white to red. could this be something with timings?
im using a completely un-modified gfx library. using preselected pins my esp has 8mb flash and 8mb psram (enabled)


![20220629_093221](https://user-images.githubusercontent.com/96497670/176455665-2584e49e-6ae6-44c7-b093-fdbdfb2b6e7f.jpg)

**Comment by moononournation** [comment, marked as answer]
I also encounter similar screen while mislead a GC9503V as ST7701V, try get update and use the new class Arduino_GC9503V_RGBPanel

**Reply by CosmicProgrammer840** [reply]
I GOT IT WORKING!!
Had to changed a few things in the display and data bus class.
board-
esp32-s3-c1 (wroom n8r8)
display -
480x480 round ips tft lcd with cpt (driver IC - st7701s)
https://www.buydisplay.com/480x480-round-ips-tft-lcd-display-2-1-inch-capactive-touch-circle-screen

//CHANGES to library
Arduino_ST7701_RGBPanel.h
lines 11&12


```
#define ST7701_TFTWIDTH 480
#define ST7701_TFTHEIGHT 480
```

Arduino_ST7701_RGBPanel.cpp
lines 14-299

```
void Arduino_ST7701_RGBPanel::begin(int32_t speed)
{
    _bus->begin(speed);

   #if 1 //
_bus->sendCommand(0xFF);
_bus->sendData(0x77);
_bus->sendData(0x01);
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x10);

_bus->sendCommand(0xC0);
_bus->sendData(0x3B);	
_bus->sendData(0x00);

_bus->sendCommand(0xC1);
_bus->sendData(0x0B);	//VBP
_bus->sendData(0x02);

_bus->sendCommand(0xC2);
_bus->sendData(0x00);
_bus->sendData(0x02);

_bus->sendCommand(0xCC);
_bus->sendData(0x10);

_bus->sendCommand(0xCD);
_bus->sendData(0x08);



_bus->sendCommand ( 0xB0); //Positive Voltage Gamma Control  
_bus->sendData ( 0x02); 
_bus->sendData ( 0x13);
_bus->sendData ( 0x1B);
_bus->sendData ( 0x0D);
_bus->sendData ( 0x10);
_bus->sendData ( 0x05);
_bus->sendData ( 0x08);
_bus->sendData ( 0x07);
_bus->sendData ( 0x07);
_bus->sendData ( 0x24);
_bus->sendData ( 0x04);
_bus->sendData ( 0x11);
_bus->sendData ( 0x0E);
_bus->sendData ( 0x2C);
_bus->sendData ( 0x33);
_bus->sendData ( 0x1D);

_bus->sendCommand ( 0xB1); //Negative Voltage Gamma Control  
_bus->sendData ( 0x05); 
_bus->sendData ( 0x13);
_bus->sendData ( 0x1B);
_bus->sendData ( 0x0D);
_bus->sendData ( 0x11);
_bus->sendData ( 0x05);
_bus->sendData ( 0x08);
_bus->sendData ( 0x07);
_bus->sendData ( 0x07);
_bus->sendData ( 0x24);
_bus->sendData ( 0x04);
_bus->sendData ( 0x11);
_bus->sendData ( 0x0E);
_bus->sendData ( 0x2C);
_bus->sendData ( 0x33);
_bus->sendData ( 0x1D);

_bus->sendCommand(0xFF);
_bus->sendData(0x77);
_bus->sendData(0x01);
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x11);

_bus->sendCommand(0xB0);   
_bus->sendData(0x5d);//5d
 
_bus->sendCommand(0xB1); 	//VCOM amplitude setting  
_bus->sendData(0x43); //43
  
_bus->sendCommand(0xB2); 	//VGH Voltage setting  
_bus->sendData(0x81);	//12V

_bus->sendCommand(0xB3);
_bus->sendData(0x80);

_bus->sendCommand(0xB5); 	//VGL Voltage setting  
_bus->sendData(0x43);	//-8.3V

_bus->sendCommand(0xB7);
_bus->sendData(0x85);

_bus->sendCommand(0xB8);
_bus->sendData(0x20);

_bus->sendCommand(0xC1);
_bus->sendData(0x78);

_bus->sendCommand(0xC2);
_bus->sendData(0x78);

_bus->sendCommand(0xD0);
_bus->sendData(0x88);

_bus->sendCommand(0xE0);
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x02);

_bus->sendCommand(0xE1);
_bus->sendData(0x03);	
_bus->sendData(0xA0);
_bus->sendData(0x00);	
_bus->sendData(0x00);
_bus->sendData(0x04);	
_bus->sendData(0xA0);
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);
_bus->sendData(0x20);
_bus->sendData(0x20);

_bus->sendCommand(0xE2);
_bus->sendData(0x00);	
_bus->sendData(0x00);	 
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);

_bus->sendCommand(0xE3);
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x11);
_bus->sendData(0x00);

_bus->sendCommand(0xE4);
_bus->sendData(0x22);
_bus->sendData(0x00);

_bus->sendCommand(0xE5);		
_bus->sendData(0x05);	
_bus->sendData(0xEC);	
_bus->sendData(0xA0);
_bus->sendData(0xA0);
_bus->sendData(0x07);	
_bus->sendData(0xEE);	
_bus->sendData(0xA0);
_bus->sendData(0xA0);
_bus->sendData(0x00);
_bus->sendData(0x00);	
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x00);	
_bus->sendData(0x00);
_bus->sendData(0x00);

_bus->sendCommand(0xE6);
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x11);
_bus->sendData(0x00);

_bus->sendCommand(0xE7);
_bus->sendData(0x22);
_bus->sendData(0x00);

_bus->sendCommand(0xE8);		
_bus->sendData(0x06);	
_bus->sendData(0xED);	
_bus->sendData(0xA0);
_bus->sendData(0xA0);
_bus->sendData(0x08);	
_bus->sendData(0xEF);	
_bus->sendData(0xA0); 
_bus->sendData(0xA0);
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x00);	
_bus->sendData(0x00);	
_bus->sendData(0x00);
_bus->sendData(0x00);

_bus->sendCommand(0xEB);
_bus->sendData(0x00); 	
_bus->sendData(0x00);
_bus->sendData(0x40);
_bus->sendData(0x40);
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x00);  

_bus->sendCommand(0xED);  
_bus->sendData(0xFF); 
_bus->sendData(0xFF);  
_bus->sendData(0xFF); 	
_bus->sendData(0xBA); 		
_bus->sendData(0x0A); 	
_bus->sendData(0xBF); 	
_bus->sendData(0x45); 	
_bus->sendData(0xFF); 
_bus->sendData(0xFF);  
_bus->sendData(0x54); 	
_bus->sendData(0xFB); 	
_bus->sendData(0xA0); 	
_bus->sendData(0xAB); 	
_bus->sendData(0xFF); 
_bus->sendData(0xFF); 
_bus->sendData(0xFF); 

_bus->sendCommand(0xEF);
_bus->sendData(0x10); 
_bus->sendData(0x0D); 
_bus->sendData(0x04); 
_bus->sendData(0x08); 
_bus->sendData(0x3F); 
_bus->sendData(0x1F);

_bus->sendCommand(0xFF);
_bus->sendData(0x77);
_bus->sendData(0x01);
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x13);

_bus->sendCommand(0xEF);
_bus->sendData(0x08);

_bus->sendCommand(0xFF);
_bus->sendData(0x77);
_bus->sendData(0x01);
_bus->sendData(0x00);
_bus->sendData(0x00);
_bus->sendData(0x00);

#if 0
_bus->_bus->sendCommand (0xFF);
WriteData (0x77);
WriteData (0x01);
WriteData (0x00);
WriteData (0x00);
WriteData (0x12);
_bus->_bus->sendCommand (0xD1);
WriteData (0x81);
WriteData (0x08);
WriteData (0x03);
WriteData (0x20);
WriteData (0x08);
WriteData (0x01);
WriteData (0xA0);
WriteData (0x01);
WriteData (0xE0);
WriteData (0xA0);
WriteData (0x01);
WriteData (0xE0);
WriteData (0x03);
WriteData (0x20);
_bus->_bus->sendCommand (0xD2);
WriteData (0x08);
#endif
/////////////////Bring up the internal test picture///////////////////////////////////


_bus->sendCommand(0x11);

delay(120);

_bus->sendCommand(0x29);

_bus->sendCommand(0x36);
_bus->sendData(0x00);

_bus->sendCommand(0x3A);
_bus->sendData(0x60);   //0x60 18bit   0x50 16bit
#endif 
```




Arduino_ESP32RGBPanel.cpp
lines 175-180
```
timings.hsync_back_porch = 18; //CHANGED
timings.hsync_front_porch = 24;//CHANGED
timings.hsync_pulse_width = 6;//CHANGED
timings.vsync_back_porch = 10;//CHANGED
timings.vsync_front_porch = 16;//CHANGED
 timings.vsync_pulse_width = 4;//CHANGED
```

![290389400_349285893991741_7084768469819362594_n](https://user-images.githubusercontent.com/96497670/176706691-d776f5cd-7366-498a-af7a-8198bb5335a2.jpg)

**Comment by moononournation** [comment]
Thanks for you code provided, Please get update and check the new code is it work:
```
Arduino_ST7701_RGBPanel *gfx = new Arduino_ST7701_RGBPanel(
  bus, 33 /* RST */, 0 /* rotation */,
  false /* IPS */, 480 /* width */, 480 /* height */,
  st7701_type3_init_operations, sizeof(st7701_type3_init_operations),
  true /* BGR */);
```

**Reply by Pestarius** [reply]
Hi,
I am trying to interface the esp32s3 to the ST7701s. I have been on it for the past three days and cannot figure it out.
The initialisation code for the ST7701s does not matter at this point. I have one from the manufacturer.
The problem is that every time I flash the esp32s3, it just loops in the reboot, so I cannot even get to the point of initialisation.
Have you got the Arduino GFX code work flashing from Arduino IDE?

**Reply by Pestarius** [reply]
This is what I got on the UART:

18:36:14.142 -> ELF file SHA256: 0000000000000000
18:36:14.142 -> 
18:36:14.142 -> Rebooting...
18:36:14.142 -> �ESP-ROM:esp32s3-20210327
18:36:14.176 -> Build:Mar 27 2021
18:36:14.176 -> rst:0xc (RTC_SW_CPU_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
18:36:14.176 -> Saved PC:0x42024552
18:36:14.176 -> SPIWP:0xee
18:36:14.176 -> mode:DIO, clock div:1
18:36:14.176 -> load:0x3fce3808,len:0x43c
18:36:14.176 -> load:0x403c9700,len:0xbec
18:36:14.176 -> load:0x403cc700,len:0x2a3c
18:36:14.176 -> SHA-256 comparison failed:
18:36:14.176 -> Calculated: dcde8d8a4817d9bf5d5d69a7247667264e4e10ac7493514868b61f5aa6146539
18:36:14.220 -> Expected: ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
18:36:14.220 -> Attempting to boot anyway...
18:36:14.220 -> entry 0x403c98d8
18:36:14.252 -> E (109) psram: PSRAM ID read error: 0x00ffffff
18:36:14.420 -> E (244) lcd_panel.rgb: esp_lcd_new_rgb_panel(150): no mem for frame buffer
18:36:14.420 -> ESP_ERROR_CHECK failed: esp_err_t 0x101 (ESP_ERR_NO_MEM) at 0x4037a7b0
18:36:14.420 -> file: "e:\Arduino\libraries\GFX_Library_for_Arduino\src\databus\Arduino_ESP32RGBPanel.cpp" line 251
18:36:14.420 -> func: uint16_t* Arduino_ESP32RGBPanel::getFrameBuffer(uint16_t, uint16_t, uint16_t, uint16_t, uint16_t, uint16_t, uint16_t, uint16_t, uint16_t, uint16_t)
18:36:14.453 -> expression: esp_lcd_new_rgb_panel(_panel_config, &_panel_handle)
18:36:14.454 -> 
18:36:14.454 -> abort() was called at PC 0x4037a7b3 on core 1
18:36:14.454 -> 
18:36:14.454 -> 
18:36:14.454 -> Backtrace:0x403776b6:0x3fce2d600x4037a7bd:0x3fce2d80 0x4037ff05:0x3fce2da0 0x4037a7b3:0x3fce2e20 0x42002f9e:0x3fce2e40 0x42003151:0x3fce2e90 0x42001838:0x3fce2ed0 0x42004aa2:0x3fce2ef0 
18:36:14.454 -> 
18:36:14.454 -> 
18:36:14.454 -> 
18:36:14.454 -> 
18:36:14.454 -> ELF file SHA256: 0000000000000000
18:36:14.454 -> 
18:36:14.454 -> Rebooting...

**Reply by moononournation** [reply]
what is your psram settings?

**Reply by CosmicProgrammer840** [reply]
those are esp32 psram errors. as the maintainer stated check your arduino ide settings and make sure they match your board specifications.
the esp32s3 you need to run this display specifically is the ESP32-S3-DevKitC-1-N8R8. with these settings:

 https://lh3.googleusercontent.com/WwbJF1pIXYo-VxETt7l9CJXM90k1tuQ6sSevhKYgMmE6zEjWQvXOvpFASqa3lFrrQ2oIppx2U_yzgqFk0ZFgmNmoPUdH5cyW_8LnkLKMn5cpePLSimf295s1XdwCO-ujyK9TbZcaEVM41Xi4hodrUQg?alr=yes

**Reply by nrjn** [reply]
Hello, @CosmicProgrammer840 @moononournation  do you think it is possible to get `ST7701s` powered display working with 
`ESP32-S3-DevKitC-1-N8 (8 MB QD, No PSRAM)`

**Reply by moononournation** [reply]
RGB display requires allocation a full framebuffer. The internal memory will limit the resolution.

**Reply by nrjn** [reply]
ok thank you

**Reply by jotisk** [reply]
Hello, is possible to reduce color bits to allow 480x480px framebuffer on S3 without PSRAM? like RGB454 or similar scheme?

**Reply by moononournation** [reply]
RGB454 cannot help reduce the size. And it is the native driver from arduino-esp32, I cannot change how it implement.

**Reply by jotisk** [reply]
OK, thank you.

**Comment by jotisk** [comment]
I was able to got ST701 working on my custom board but I have problems with some glitches when repeatedly writing to display. GIF animation works first few frames and then looks like broken hsync. after while it is correct again. Same with hello world example, first few texts are correct then is image distorted for a while and sometimes fixes itself. 
Any idea where to start?
`#include <stdio.h>
#include <freertos/FreeRTOS.h>
#include <freertos/task.h>
#include "sdkconfig.h"
#include <Arduino.h>
#include <SPI.h>
#include <Arduino_GFX_Library.h>
#include "image.h"
//#include <Adafruit_GFX.h>

#define GIF_FILENAME "/power.gif"
#define ESP32

#include <SPIFFS.h>
#include "GifClass.h"
static GifClass gifClass;

/* More data bus class: https://github.com/moononournation/Arduino_GFX/wiki/Data-Bus-Class */
Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
  5 /* CS */, 4 /* SCK */, 2 /* SDA */,
  7 /* DE */, 15 /* VSYNC */, 16 /* HSYNC */, 6 /* PCLK */,
  8 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
  41 /* G0 */, 40 /* G1 */, 39 /* G2 */, 38 /* G3 */, 45 /* G4 */, 46 /* G5 */,
  14 /* B0 */, 9 /* B1 */, 10 /* B2 */, 11 /* B3 */, 42 /* B4 */
);

/* More display class: https://github.com/moononournation/Arduino_GFX/wiki/Display-Class */
Arduino_ST7701_RGBPanel *gfx = new Arduino_ST7701_RGBPanel(
  bus, GFX_NOT_DEFINED /* RST */, 0 /* rotation */,
  false /* IPS */, 480 /* width */, 480 /* height */,
  st7701_type3_init_operations, sizeof(st7701_type3_init_operations),
  true /* BGR */,24,6,18,16,4,10);

#define GFX_BL 12 // default backlight pin, you may replace DF_GFX_BL to actual backlight pin

#define USE_ESP_IDF_LOG 

/*******************************************************************************
 * End of Arduino_GFX setting
 ******************************************************************************/



void setup(void)
{

Serial.begin(115200);
Serial.print("START");

    gfx->begin();
    SPIFFS.begin();
    gfx->fillScreen(BLACK);

  gfx->setCursor(100, 100);
    gfx->setTextColor(RED);
    gfx->setTextSize(4,4);
    gfx->println("Hello World!");
    delay(1000); // 1 second
}

void loop()
{

File gifFile = SPIFFS.open(GIF_FILENAME, "r");

if (!gifFile || gifFile.isDirectory())
  {
    Serial.println(F("ERROR: open gifFile Failed!"));
    gfx->println(F("ERROR: open gifFile Failed!"));
  }
  else
  {
    // read GIF file header
    gd_GIF *gif = gifClass.gd_open_gif(&gifFile);
    if (!gif)
    {
      Serial.println(F("gd_open_gif() failed!"));
    }
    else
    {
      uint8_t *buf = (uint8_t *)malloc(gif->width * gif->height);
      if (!buf)
      {
        Serial.println(F("buf malloc failed!"));
      }
      else
      {
        int16_t x = 50;// (gfx->width() - gif->width) / 2;
        int16_t y = 50;//(gfx->height() - gif->height) / 2;

        Serial.println(F("GIF video start"));
        int32_t t_fstart, t_delay = 0, t_real_delay, delay_until;
        int32_t res = 1;
        int32_t duration = 0, remain = 0;
        while (res > 0)
        {
          t_fstart = millis();
          t_delay = gif->gce.delay * 10;
          res = gifClass.gd_get_frame(gif, buf);
          if (res < 0)
          {
            Serial.println(F("ERROR: gd_get_frame() failed!"));
            break;
          }
          else if (res > 0)
          {
            gfx->drawIndexedBitmap(x, y, buf, gif->palette->colors, gif->width, gif->height);

            t_real_delay = t_delay - (millis() - t_fstart);
            duration += t_delay;
            remain += t_real_delay;
            delay_until = millis() + t_real_delay;
            while (millis() < delay_until)
            {
              delay(1);
            }
          }
        }
        Serial.println(F("GIF video end"));
        Serial.print(F("duration: "));
        Serial.print(duration);
        Serial.print(F(", remain: "));
        Serial.print(remain);
        Serial.print(F(" ("));
        Serial.print(100.0 * remain / duration);
        Serial.println(F("%)"));

        gifClass.gd_close_gif(gif);
        free(buf);
      }
    }
  }

Serial.println(F("LOOP END"));
    delay(1000); // 1 second
}`

Another question, is possible to speed up gif framebuffer write? with earth gif animation I can see how is frame writing to framebuffer on display (lines are flickering like on your video) or it is limitation of psram performance?

Thank you very much for support.

**Reply by moononournation** [reply]
It have a big change on RGBPanel and release soon. You may get update and follow the latest PDQGraphicstest.ino to test again.

**Reply by jotisk** [reply]
perfect, thank you for info, I will watch for update.

**Reply by jotisk** [reply]
May I ask you for hint how to init RGB panel with new library 1.3.1? 
old code
` /* More data bus class: https://github.com/moononournation/Arduino_GFX/wiki/Data-Bus-Class */
Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
  5 /* CS */, 4 /* SCK */, 2 /* SDA */,
  7 /* DE */, 15 /* VSYNC */, 16 /* HSYNC */, 6 /* PCLK */,
  8 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
  41 /* G0 */, 40 /* G1 */, 39 /* G2 */, 38 /* G3 */, 45 /* G4 */, 46 /* G5 */,
  14 /* B0 */, 9 /* B1 */, 10 /* B2 */, 11 /* B3 */, 42 /* B4 */
);

/* More display class: https://github.com/moononournation/Arduino_GFX/wiki/Display-Class */
Arduino_ST7701_RGBPanel *gfx = new Arduino_ST7701_RGBPanel(
  bus, GFX_NOT_DEFINED /* RST */, 0 /* rotation */,
  false /* IPS */, 480 /* width */, 480 /* height */,
  st7701_type3_init_operations, sizeof(st7701_type3_init_operations),
  true /* BGR */,24,6,18,16,4,10);`


I tried new one, but data are somehow not sent:
`/* More data bus class: https://github.com/moononournation/Arduino_GFX/wiki/Data-Bus-Class */
Arduino_ESP32RGBPanel *panel = new Arduino_ESP32RGBPanel(
 // 5 /* CS */, 4 /* SCK */, 2 /* SDA */,
  7 /* DE */, 15 /* VSYNC */, 16 /* HSYNC */, 6 /* PCLK */,
  8 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
  41 /* G0 */, 40 /* G1 */, 39 /* G2 */, 38 /* G3 */, 45 /* G4 */, 46 /* G5 */,
  14 /* B0 */, 9 /* B1 */, 10 /* B2 */, 11 /* B3 */, 42 /* B4 */,
  1,24,6,18,
  1,16,4,10
);

Arduino_DataBus *bus = new Arduino_ESP32SPI(-1, 5, 4, 2, -1);


/* More display class: https://github.com/moononournation/Arduino_GFX/wiki/Display-Class */
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
  480,480,panel,0,true,bus,-1,st7701_type3_init_operations, sizeof(st7701_type3_init_operations));`

Thank  you very much.

**Reply by moononournation** [reply]
still developing, please see PDQGraphicstest for more examples

**Reply by jotisk** [reply]
great, thanks.
